### DISCUSSION TITLE
GitHub PR filter not working for review:approved

### DISCUSSION THREAD

**Discussion by catenacyber** [community/community#55826] [Pull Requests]
### Select Topic Area

Bug

### Body

https://github.com/OISF/suricata/pulls?q=is%3Aopen+is%3Apr+review%3Aapproved does not show https://github.com/OISF/suricata/pull/8891

Also not that https://github.com/OISF/suricata/pull/8891 does not show in other `review` filters like 
https://github.com/OISF/suricata/pulls?q=is%3Aopen+is%3Apr+review%3Anone
or https://github.com/OISF/suricata/pulls?q=is%3Aopen+is%3Apr+review%3Achanges-requested

From https://docs.github.com/en/issues/tracking-your-work-with-issues/filtering-and-searching-issues-and-pull-requests the filter should show pull requests that a reviewer has approved, and that is the case.

Also note that this feature works on master-6.0.x branch, so it may be related to the CODEOWNERS file somehow

**Comment by catenacyber** [comment]
`gh pr view --json reviewDecision 8891` gives 
```
{
  "reviewDecision": ""
}
```

while `gh pr view --json reviewDecision 8897` gives as expected
```
{
  "reviewDecision": ""
}
```

Or `gh pr view --json reviewDecision 8870`gives as expected
```
{
  "reviewDecision": "CHANGES_REQUESTED"
}
```

And `gh pr view --json reviews 8891` gives
```
{
  "reviews": [
    {
      "id": "PRR_kwDOAFKNus5VVU18",
      "author": {
        "login": "victorjulien"
      },
      "authorAssociation": "MEMBER",
      "body": "",
      "submittedAt": "2023-05-17T21:10:43Z",
      "includesCreatedEdit": false,
      "reactionGroups": [],
      "state": "APPROVED",
      "commit": {
        "oid": "5fb0b3b8cb471f5518398426750ec2caad2e47a1"
      }
    },
    {
      "id": "PRR_kwDOAFKNus5VYDkb",
      "author": {
        "login": "jasonish"
      },
      "authorAssociation": "MEMBER",
      "body": "",
      "submittedAt": "2023-05-18T09:32:14Z",
      "includesCreatedEdit": false,
      "reactionGroups": [],
      "state": "APPROVED",
      "commit": {
        "oid": "5fb0b3b8cb471f5518398426750ec2caad2e47a1"
      }
    },
    {
      "id": "PRR_kwDOAFKNus5VYEZG",
      "author": {
        "login": "jlucovsky"
      },
      "authorAssociation": "MEMBER",
      "body": "",
      "submittedAt": "2023-05-18T09:34:35Z",
      "includesCreatedEdit": false,
      "reactionGroups": [],
      "state": "APPROVED",
      "commit": {
        "oid": "5fb0b3b8cb471f5518398426750ec2caad2e47a1"
      }
    }
  ]
}
```

**Comment by catenacyber** [comment]
`gh pr list --json number,reviewDecision --search "state:open type:pr -review:none" | jq '.[] | select(.reviewDecision=="")'` is the hack I found to get approved PRs

Gets now
```
{
  "number": 8910,
  "reviewDecision": ""
}
{
  "number": 8907,
  "reviewDecision": ""
}
{
  "number": 8878,
  "reviewDecision": ""
}
{
  "number": 8714,
  "reviewDecision": ""
}
```

**Comment by bselwe** [comment]
Is there any update on this? I'm facing the same issue in one of [joinbuzz repositories](https://github.com/joinbuzz).

**Reply by catenacyber** [reply]
Yesterday, this was fixed for Suricata... not sure how...

**Reply by bselwe** [reply]
@catenacyber Maybe `review:approved` means that the pull request has been approved by the code owner? 🤷🏻 But this is not stated in the documentation.

**Reply by catenacyber** [reply]
We tried that, and it was not enough

**Reply by bselwe** [reply]
Interesting, I'm trying to group PRs in either "Approved", "Changes Requested" or "Other" sections. The PR with approvals seems to be displayed in the "Other" section when using the following filter: `-review:approved -review:changes_requested` (but not in the "Approved" section).

**Reply by catenacyber** [reply]
I think you cannot have two negations about `review`

`review:none` should be the missing one, no ?

**Reply by bselwe** [reply]
@catenacyber That's what I tried before, but when a PR is approved by a non-codeowner, then it has a review - so it won't be shown in `review:none` filter. It is also not shown in `review:approved` filter as observed in this issue.

Using `-review:approved -review:changes_requested` seems to do the trick as it returns PRs that are neither approved nor changes requested. Why do you think that having two negations won't work?

**Reply by catenacyber** [reply]
So, looks like a different problem than the one I reported which seemed fixed this week.

> Why do you think that having two negations won't work?

I remember trying and failing... But maybe that got fixed since

**Reply by bselwe** [reply]
I am using [GitHub Search API](https://docs.github.com/en/rest/search/search?apiVersion=2022-11-28#search-issues-and-pull-requests), maybe there's a difference in results when using CLI? 🤷🏻

**Reply by bselwe** [reply]
You can try running your search query [here on GitHub](https://github.com/search?q=is%3Apr%20is%3Aopen&type=pullrequests) if you'd like to compare results.

**Comment by JustinTRoss** [comment]
Has anyone been able to figure out a work-around for this?

_(A GitHub search filter whose results include PRs which have been approved by at least one user, though which may or may not have been approved by all required reviewers)_

**Comment by ashwinstar** [comment]
> A GitHub search filter whose results include PRs which have been approved by at least one user, though which may or may not have been approved by all required reviewers

Do we have way to to accomplish this now in github?

**Reply by ashwinstar** [reply]
I didn't find a syntax or capability which allows to do the required. Hence requesting to enhance the functionality, or if I am missing on existing capability to accomplish the goal then please provide me with the query for the same.

**Comment by strokirk** [comment]
If anyone from Github happens to be reading this, a suggestion could be to maybe add a filter for ”review-approvals:>X”, to more easily handle the case of organizations which have different amount of required reviewers, or when there is a mix of approvals and change requests.

Though it would also be good to be able to filter for ”approved by at least one person, with no rejections”.

**Comment by krassowski** [comment]
This is quite annoying because it just stopped working some time ago. https://github.com/jupyterlab/jupyterlab/pulls?q=is%3Apr+is%3Aopen+review%3Aapproved does not return any approved PRs:

![image](https://github.com/user-attachments/assets/5df97333-60b3-4d4a-a6d9-d31cd75be794)

but  there ARE many approved open PRs like this one:

![image](https://github.com/user-attachments/assets/45b61f89-5cff-43d4-9c1a-061961801a1e)

but I don't know which ones are those

**Comment by krassowski** [comment]
This is still broken in 2026 :(

**Comment by krassowski** [comment]
There is more users reporting this on Stack Overflow: https://stackoverflow.com/questions/75637877/github-filtering-not-showing-approved-reviews

**Comment by tgross35** [comment]
I wrote this up in https://github.com/orgs/community/discussions/144580 as well, before having seen this discussion.
